### PR TITLE
Replace update.php with update.sh and Dockerfile templates

### PR DESCRIPTION
**Pull request by J0WI**

### CODE REVIEW

File: update.sh [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+phpVersions=( "$@" )
+if [ ${#phpVersions[@]} -eq 0 ]; then
+	phpVersions=( php*.*/ )
+fi
+phpVersions=( "${phpVersions[@]%/}" )
+
+current="$(curl -fsSL 'https://downloads.joomla.org/api/v1/latest/cms' | jq -r '.branches[3].version')"
+urlVersion=$(echo $current | sed -e 's/\./-/g')
+sha1="$(curl -fsSL "https://downloads.joomla.org/api/v1/signatures/cms/$urlVersion"  | jq -r '.files[9].sha1')"
```
Comment (mbabker) [inline update.sh]: The files count here won't always be reliable.  Our .0 releases only have 6 total packages, .1 releases have 9 packages, and .2+ have 12.

Comment (J0WI) [reply update.sh]: Is there a simpler way than the for loop from `update.php`?

Comment (mbabker) [reply update.sh]: If you can find a way to get the node with `"filename":"Joomla_$current-Stable-Full_Package.tar.bz2"` more simply then that'd be it.  Or we modify our API response to use filenames as keys (not sure how well that works with the [OpenAPI specification](https://github.com/joomla/downloads.joomla.org/blob/master/libraries/api/application/open_api.json) used for the API though).

Comment (J0WI) [reply update.sh]: Okay, that was a bit tricky 😄

**Comment by J0WI** [comment]
You may also give write access to @docker-library-bot

**Comment by mbabker** [comment]
I might have prematurely merged this one...

```sh
[jenkins@joomla-jen docker-joomla]$ ./update.sh 
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!5.6!g' -e 's!%%VARIANT%%!apache!g' -e 's!%%VARIANT_EXTRAS%%!\n# Enable Apache Rewrite Module\nRUN a2enmod rewrite\n!g' -e 's!%%APCU_VERSION%%!4.0.11!g' -e 's!%%MEMCACHED_VERSION%%!2.2.0!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!apache2-foreground!g' Dockerfile-debian.template
+ cp -a docker-entrypoint.sh php5.6/apache/docker-entrypoint.sh
+ cp -a makedb.php php5.6/apache/makedb.php
+ '[' php5.6 = php7.2 ']'
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!5.6!g' -e 's!%%VARIANT%%!fpm!g' -e 's!%%VARIANT_EXTRAS%%!!g' -e 's!%%APCU_VERSION%%!4.0.11!g' -e 's!%%MEMCACHED_VERSION%%!2.2.0!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!php-fpm!g' Dockerfile-debian.template
+ cp -a docker-entrypoint.sh php5.6/fpm/docker-entrypoint.sh
+ cp -a makedb.php php5.6/fpm/makedb.php
+ '[' php5.6 = php7.2 ']'
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!5.6!g' -e 's!%%VARIANT%%!fpm-alpine!g' -e 's!%%VARIANT_EXTRAS%%!!g' -e 's!%%APCU_VERSION%%!4.0.11!g' -e 's!%%MEMCACHED_VERSION%%!2.2.0!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!php-fpm!g' Dockerfile-alpine.template
+ cp -a docker-entrypoint.sh php5.6/fpm-alpine/docker-entrypoint.sh
+ cp -a makedb.php php5.6/fpm-alpine/makedb.php
+ '[' php5.6 = php7.2 ']'
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!7.0!g' -e 's!%%VARIANT%%!apache!g' -e 's!%%VARIANT_EXTRAS%%!\n# Enable Apache Rewrite Module\nRUN a2enmod rewrite\n!g' -e 's!%%APCU_VERSION%%!5.1.11!g' -e 's!%%MEMCACHED_VERSION%%!3.0.4!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!apache2-foreground!g' Dockerfile-debian.template
+ cp -a docker-entrypoint.sh php7.0/apache/docker-entrypoint.sh
+ cp -a makedb.php php7.0/apache/makedb.php
+ '[' php7.0 = php7.2 ']'
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!7.0!g' -e 's!%%VARIANT%%!fpm!g' -e 's!%%VARIANT_EXTRAS%%!!g' -e 's!%%APCU_VERSION%%!5.1.11!g' -e 's!%%MEMCACHED_VERSION%%!3.0.4!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!php-fpm!g' Dockerfile-debian.template
+ cp -a docker-entrypoint.sh php7.0/fpm/docker-entrypoint.sh
+ cp -a makedb.php php7.0/fpm/makedb.php
+ '[' php7.0 = php7.2 ']'
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!7.0!g' -e 's!%%VARIANT%%!fpm-alpine!g' -e 's!%%VARIANT_EXTRAS%%!!g' -e 's!%%APCU_VERSION%%!5.1.11!g' -e 's!%%MEMCACHED_VERSION%%!3.0.4!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!php-fpm!g' Dockerfile-alpine.template
+ cp -a docker-entrypoint.sh php7.0/fpm-alpine/docker-entrypoint.sh
+ cp -a makedb.php php7.0/fpm-alpine/makedb.php
+ '[' php7.0 = php7.2 ']'
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!7.1!g' -e 's!%%VARIANT%%!apache!g' -e 's!%%VARIANT_EXTRAS%%!\n# Enable Apache Rewrite Module\nRUN a2enmod rewrite\n!g' -e 's!%%APCU_VERSION%%!5.1.11!g' -e 's!%%MEMCACHED_VERSION%%!3.0.4!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!apache2-foreground!g' Dockerfile-debian.template
+ cp -a docker-entrypoint.sh php7.1/apache/docker-entrypoint.sh
+ cp -a makedb.php php7.1/apache/makedb.php
+ '[' php7.1 = php7.2 ']'
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!7.1!g' -e 's!%%VARIANT%%!fpm!g' -e 's!%%VARIANT_EXTRAS%%!!g' -e 's!%%APCU_VERSION%%!5.1.11!g' -e 's!%%MEMCACHED_VERSION%%!3.0.4!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!php-fpm!g' Dockerfile-debian.template
+ cp -a docker-entrypoint.sh php7.1/fpm/docker-entrypoint.sh
+ cp -a makedb.php php7.1/fpm/makedb.php
+ '[' php7.1 = php7.2 ']'
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!7.1!g' -e 's!%%VARIANT%%!fpm-alpine!g' -e 's!%%VARIANT_EXTRAS%%!!g' -e 's!%%APCU_VERSION%%!5.1.11!g' -e 's!%%MEMCACHED_VERSION%%!3.0.4!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!php-fpm!g' Dockerfile-alpine.template
+ cp -a docker-entrypoint.sh php7.1/fpm-alpine/docker-entrypoint.sh
+ cp -a makedb.php php7.1/fpm-alpine/makedb.php
+ '[' php7.1 = php7.2 ']'
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!7.2!g' -e 's!%%VARIANT%%!apache!g' -e 's!%%VARIANT_EXTRAS%%!\n# Enable Apache Rewrite Module\nRUN a2enmod rewrite\n!g' -e 's!%%APCU_VERSION%%!5.1.11!g' -e 's!%%MEMCACHED_VERSION%%!3.0.4!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!apache2-foreground!g' Dockerfile-debian.template
+ cp -a docker-entrypoint.sh php7.2/apache/docker-entrypoint.sh
+ cp -a makedb.php php7.2/apache/makedb.php
+ '[' php7.2 = php7.2 ']'
+ sed -e /libmcrypt-dev/d -e /mcrypt/d -i php7.2/apache/Dockerfile
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!7.2!g' -e 's!%%VARIANT%%!fpm!g' -e 's!%%VARIANT_EXTRAS%%!!g' -e 's!%%APCU_VERSION%%!5.1.11!g' -e 's!%%MEMCACHED_VERSION%%!3.0.4!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!php-fpm!g' Dockerfile-debian.template
+ cp -a docker-entrypoint.sh php7.2/fpm/docker-entrypoint.sh
+ cp -a makedb.php php7.2/fpm/makedb.php
+ '[' php7.2 = php7.2 ']'
+ sed -e /libmcrypt-dev/d -e /mcrypt/d -i php7.2/fpm/Dockerfile
+ sed -r -e 's!%%VERSION%%!3.8.8!g' -e 's!%%SHA1%%!d905327bee0cd5b4806ef34f74cbd5214fc9da2c!g' -e 's!%%PHP_VERSION%%!7.2!g' -e 's!%%VARIANT%%!fpm-alpine!g' -e 's!%%VARIANT_EXTRAS%%!!g' -e 's!%%APCU_VERSION%%!5.1.11!g' -e 's!%%MEMCACHED_VERSION%%!3.0.4!g' -e 's!%%REDIS_VERSION%%!3.1.6!g' -e 's!%%CMD%%!php-fpm!g' Dockerfile-alpine.template
+ cp -a docker-entrypoint.sh php7.2/fpm-alpine/docker-entrypoint.sh
+ cp -a makedb.php php7.2/fpm-alpine/makedb.php
+ '[' php7.2 = php7.2 ']'
+ sed -e /libmcrypt-dev/d -e /mcrypt/d -i php7.2/fpm-alpine/Dockerfile
```

**Comment by J0WI** [comment]
Seems to run without any failures, doesn't it?

**Comment by mbabker** [comment]
Guess it does (after modifying update.sh to point to old version just to see what happens).  Just wasn't expecting all of that output honestly.

**Comment by J0WI** [comment]
It's just this line to see what's going on:
https://github.com/joomla/docker-joomla/blob/c74ab1bb051dbddcbe79003db8f37861346fc3cf/update.sh#L57

**Comment by mbabker** [comment]
I did mention somewhere my shell scripting knowledge isn't as strong as I'd like it to be 😆

**Pull request merged**